torch/utils/data/dataloader.py

```diff
@@ -228,9 +228,9 @@ def __init__(self, dataset: Dataset[T_co], batch_size: Optional[int] = 1,
             # option. If this turns out to be useful in future, we can re-enable
             # this, and support custom samplers that specify the assignments to
             # specific workers.
-            if isinstance(dataset, IterDataPipe):
+            if isinstance(dataset, IterDataPipe) and shuffle is not None:
                 dataset = torch.utils.data.graph_settings.apply_shuffle_settings(dataset, shuffle=shuffle)
-            elif shuffle is not False:
+            elif shuffle:
```

Comment:
How about we convert `shuffle` back to `False` after the logic for `IterDataPipe` to preserve BC?